### PR TITLE
Two modules independent of audacity

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: *(direct link to the issue)*

Two of the import-export modules can break their dependency on the Audacity executable, with only two .cpp files moved from Audacity to a new, "red" library, lib-preference-pages.

If these ideas are pursued, that library might turn "green." https://github.com/audacity/audacity/discussions/5628

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by Paul-Licameli** [comment]
Please give a quick look, guys.  Like it?  Nominate it for the 3.5 board.

File: modules/mod-mp3/ExportMP3.cpp [hunk 1]
```diff
@@ -1702,9 +1701,6 @@ bool MP3ExportProcessor::Initialize(AudacityProject& project,
    context.channels = channels;
 
    int rate = lrint(sampleRate);
-#ifndef DISABLE_DYNAMIC_LOADING_LAME
```
Comment (crsib) [inline modules/mod-mp3/ExportMP3.cpp:1705]: It is really odd that this code is still here, provided that Audacity links against lame directly for quite a few years. I think that we need to revise ExportMP3 code.

Comment (Paul-Licameli) [reply modules/mod-mp3/ExportMP3.cpp:1705]: That could be separately addressed of course.  What do you think of this PR otherwise?

**Review by crsib — APPROVED** [review]
Provided that this PR is quite simple I think we should take it

**Comment by Paul-Licameli** [comment]
I showed you what's possible in the modules dependency graph with a little work.

I moved only the minimal necessary files into lib-preference-pages for this.

If more pages might be easily moved into the library, perhaps they should, but then the library will serve miscellaneous purposes, grouping rather unrelated preferences together.

**Comment by Paul-Licameli** [comment]
This PR was approved but forgotten.  I rebased it and fixed conflicts.  Please reconsider.

**Review by grliszas14 — APPROVED** [review]

**Comment by dozzzzer** [comment]
Tested

**Pull request merged**